search-app/src/cmr/search/services/humanizers/humanizer_report_service.clj

```diff
@@ -95,3 +105,88 @@
                        "concat humanized rows:" t3))))]
       (info "Create report " t4)
       (str string-writer))))
+
+(defn safe-generate-humanizers-report-csv
+  "This convenience function wraps the report generator in a try/catch."
+  [context]
+  (try
+    (generate-humanizers-report-csv context)
+    (catch Exception e
+      (warn (.getMessage e) "Returning empty report."))))
+
+(def report-cache-key
+  "The key used to store the humanizer report cache in the system cache map."
+  :report-cache)
+
+(defn get-cache
+  "A utility function for returning the system report cache instance."
+  [context]
+  (get-in context [:system :caches report-cache-key]))
+
+(defn read-report
+  "A convenience function wrapping the cache object to retreive report
+  data."
+  [context]
+  (.get-value (get-cache context)
+              csv-report-cache-key))
+
+(defn write-report
+  "A convenience function wrapping the cache object to retreive report
+  data."
+  [context]
+  (.set-value (get-cache context)
+              csv-report-cache-key
+              (safe-generate-humanizers-report-csv context)))
+
+(defconfig report-cache-consistent-timeout-seconds
+  "The number of seconds between when the report cache should check with cubby
+  for data consistency."
+  {:default (* 60 60 6) ; every 6 hours
```

Comment:
I think we may still run into problems here if I understand this correctly. If it has been more than six hours since someone has generated the report then a request will cause the report to be generated, correct?. This will still suffer from the timeout issue since generation isn't any faster. The workaround is to wait for five or six minutes and make the request again, by which time it should be cached, I think. Which is better than what we have now.